### PR TITLE
Added a level script function to easily set a skybox for each area

### PR DESCRIPTION
**Pull request by Mineqwerty**
In your level script add the function
CHANGE_AREA_SKYBOX(Area ID, Skybox Segment Start, Skybox Segment End) .

### CODE REVIEW

**Comment by FazanaJ** [comment]
Small thing missed, puppycam volumes inside `puppycam.h` rely on spot 0x3D

Consider (as in do it 🔫) changing that to 0x3E, so it will continue to work after the changes

File: src/game/area.c [hunk 1]
```diff
@@ -225,7 +225,7 @@ void load_area(s32 index) {
     if (gCurrentArea == NULL && gAreaData[index].unk04 != NULL) {
         gCurrentArea = &gAreaData[index];
         gCurrAreaIndex = gCurrentArea->index;
-
+
```
Comment (someone2639) [inline src/game/area.c:228]: style issue 1

File: src/game/area.c [hunk 2]
```diff
@@ -236,7 +236,7 @@ void load_area(s32 index) {
         }
 
         load_obj_warp_nodes();
-        geo_call_global_function_nodes(&gCurrentArea->unk04->node, GEO_CONTEXT_AREA_LOAD);
+        geo_call_global_function_nodes(&gCurrentArea->unk04->node, GEO_CONTEXT_AREA_LOAD);
```
Comment (someone2639) [inline src/game/area.c:239]: style issue 2

File: src/game/area.c [hunk 3]
```diff
@@ -259,6 +259,9 @@ void load_mario_area(void) {
         gCurrentArea->flags |= 0x01;
         spawn_objects_from_info(0, gMarioSpawnInfo);
     }
+    if (gAreaSkyboxStart[gCurrAreaIndex-1]) {
+        load_segment_decompress(0x0A, gAreaSkyboxStart[gCurrAreaIndex-1], gAreaSkyboxEnd[gCurrAreaIndex-1]);
+        }
```
Comment (someone2639) [inline src/game/area.c:264]: style issue 3

File: src/game/game_init.c [hunk 4]
```diff
@@ -66,6 +66,8 @@ u8 gIsConsole;
 #ifdef WIDE
 u8 gWidescreen;
 #endif
+u8 *gAreaSkyboxStart[7];
```
Comment (someone2639) [inline src/game/game_init.c:69]: is there an area count define you can use to size these instead of hardcoding to 7?

**Review by someone2639 — CHANGES_REQUESTED** [review]

File: src/game/game_init.c (hunk 4 above)
Comment (someone2639) [reply src/game/game_init.c:69]: same thing in the header

File: src/engine/level_script.c [hunk 5]
```diff
@@ -304,11 +304,22 @@ static void level_cmd_load_yay0_texture(void) {
...
+    gAreaSkyboxEnd[areaCheck-1] = CMD_GET(void *, 8);
+    sCurrentCmd = CMD_NEXT;
+}
+
 static void level_cmd_init_level(void) {
     init_graph_node_start(NULL, (struct GraphNodeStart *) &gObjParentGraphNode);
     clear_objects();
     clear_areas();
     main_pool_push_state();
+    for (u8 clearPointers; clearPointers < 8; clearPointers++) {
```
Comment (someone2639) [inline src/engine/level_script.c:319]: no init value is undefined behavior 😡

**Review by someone2639 — CHANGES_REQUESTED** [review]
this one probably actually falls under Request Changes

**Review by someone2639 — APPROVED** [review]
issues are only minor now (something i can fix in the next epic pr)

**Pull request merged**